docs/source/blueprints/BP003.rst

```diff
@@ -0,0 +1,573 @@
+iteration.
+
+Notice that some information, such as the ``PASS`` for the second test,
+will come from the "result" given in a status message from the task
+itself. Some other status, such as the ``CANCEL`` status for the
+first test will not come from a status message received, but from a
+realization of the actual management of the task execution. It's
+expected to other information will also have to be inferred, and
+"filled in" by the nrunner plugin implementation.
+
+In the end, it's expected that results similar to this would be presented::
+
+    JOB ID     : f59bd40b8ac905864c4558dc02b6177d4f422ca3
+    JOB LOG    : /home/cleber/avocado/job-results/job-2020-05-20T17.58-f59bd40/job.log
+     (1/2) tests.py:Test.test_1: CANCEL (0 s)
+     (1/2) tests.py:Test.test_2: PASS (2.56 s)
+    RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
+    JOB TIME   : 0.19 s
+    JOB HTML   : /home/cleber/avocado/job-results/job-2020-05-20T17.58-f59bd40/results.html
+
+Notice that Task #2 may show up before Task #1.  There may be issues
+associated with the current UI to deal with regarding out of order
+task status updates.
+
+Implementation Example
+**********************
+
+The following implementation example uses random sleeps and random
+(but biased) results from operations expected to happen on different
+phases, to simulate the behavior of real tasks.
+
+The enforcement of some artifical limits (such as the number of tasks
+``TASK_STARTED``) is also exemplified.  As a general rule, all tasks
+are attempted to be moved further into their life-cycle and a number
+of "workers" doing that should not conflict with each other.
+
+This implementation uses Python's :mod:`asyncio` library very crudely.
+The final implementation may use other tools, such as a
+:class:`asyncio.Queue` instead of plain lists with a
+:class:`asyncio.Lock`.  It may also use individual :class:`Tasks
+<asyncio.Task>` for each work in each phase.
+
+.. literalinclude:: ../../../examples/nrunner/task-state-machine.py
+
+Backwards Compatibility
+***********************
+
+The compatibility of the resulting Job compatible runner
+implementation with the current runner is to be verified by running
+the same set of "Job API feature tests", but with this runner selected
+instead.
+
+There are no compatibility issues with the previous versions of
+itself, or with the non-Job compatible ``nrun`` implementation.
+
+Security Implications
+*********************
+
+None that we can determine at this point.
+
+How to Teach This
+*****************
+
+The distinctive features that the N(ext) Runner provides should be
+properly documented.
+
+Users should not be required to learn about the N(ext) Runner features
+to use it just as an alternative to the current runner implementation.
+
+Related Issues
+**************
+
+Current issues that are expected to be solved when this blueprint is
+implemented:
+
+#. Have a passive Task Status collection server implementation.
```

Comment:
Just to be picky: A "passive task status collection server implementation" is not an issue, right?

Reply:
Well, I was trying to follow the model on BP001, such as:

```
#. Create a simple but effective type system for configuration files and
   argument options.
```

That is, with issues that will be created (in GH) and solved when this BP is considered implemented.

Reply:
Ohh I see. Ok, makes sense. sorry about the noise.

Reply:
Maybe this become more clear if it is written just like the issue: "Implement the passive Task Status collection server."